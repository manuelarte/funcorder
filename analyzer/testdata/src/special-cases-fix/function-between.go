package special_cases_fix

// FunctionBetweenStruct comment
type FunctionBetweenStruct struct {
}

// Open comment
func Open() (*FunctionBetweenStruct, error) {
	return &FunctionBetweenStruct{}, nil
}

// isOpen comment
func (f *FunctionBetweenStruct) isOpen() bool { // want `unexported method "isOpen" for struct "FunctionBetweenStruct" should be placed after the exported method "Close"`
	return true
}

// cannotOpenFileError comment
type cannotOpenFileError struct {
	err error
}

// Unwrap comment
func (e *cannotOpenFileError) Unwrap() error {
	return e.err
}

func (f *FunctionBetweenStruct) Close() {
}
