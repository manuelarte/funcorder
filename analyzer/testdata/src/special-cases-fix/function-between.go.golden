package special_cases_fix

// FunctionBetweenStruct comment
type FunctionBetweenStruct struct {
}

// Open comment
func Open() (*FunctionBetweenStruct, error) {
	return &FunctionBetweenStruct{}, nil
}

func (f *FunctionBetweenStruct) Close() {
}

// isOpen comment
func (f *FunctionBetweenStruct) isOpen() bool {
	return true
}

// cannotOpenFileError comment
type cannotOpenFileError struct {
	err error
}

// Unwrap comment
func (e *cannotOpenFileError) Unwrap() error {
	return e.err
}
